src/helper/selection-tools/scale-tool.js

```diff
@@ -77,33 +78,36 @@ class ScaleTool {
         const delta = point.subtract(this.lastPoint);
         this.lastPoint = point;
 
-        const modOrigSize = this.origSize;
-
         if (event.modifiers.alt) {
             this.centered = true;
             this.itemGroup.position = this.origCenter;
             this.pivot = this.origCenter;
-            this.modOrigSize = this.origSize * 0.5;
         } else {
             if (this.centered) {
                 // Reset position if we were just in alt
                 this.centered = false;
                 this.itemGroup.scale(1 / this.lastSx, 1 / this.lastSy, this.pivot);
+                if (this.selectionAnchor) {
+                    this.selectionAnchor.scale(this.lastSx, this.lastSy);
+                }
                 this.lastSx = 1;
                 this.lastSy = 1;
             }
             this.pivot = this.origPivot;
         }
 
         this.corner = this.corner.add(delta);
-        const size = this.corner.subtract(this.pivot);
+        let size = this.corner.subtract(this.pivot);
+        if (event.modifiers.alt) {
```

Comment:
Hmmm this seems like an unrelated feature that it'd be good to put in separate PRs. Please try to keep PRs feature specific/small as possible to make it easier to review.